specutils/extern/spectralcoord/__init__.py

```diff
@@ -0,0 +1,49 @@
+###############################################################################
+# This folder contains a copy of the SpectralQuantity and SpectralCoord classes
+# from astropy core. If astropy>=4.1 is available, the SpectralCoord class is
+# used directly from there, otherwise it is imported from the folder here. No
+# changes should be made directly here, instead changes should be made to the
+# core astropy package, and files copied back here. The only change that should
+# be made is to change the import of SpectralQuantity from an absolute to a
+# relative import.
+#
+# The license of the original code is:
+#
+# Copyright (c) 2011-2017, Astropy Developers
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice, this
+#   list of conditions and the following disclaimer in the documentation and/or
+#   other materials provided with the distribution.
+# * Neither the name of the Astropy Team nor the names of its contributors may be
+#   used to endorse or promote products derived from this software without
+#   specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+from distutils.version import LooseVersion
+
+from astropy import __version__ as astropy_version
+
+if LooseVersion(astropy_version) >= '4.1':
+    from astropy.coordinates import SpectralCoord
+else:
+    from .spectral_coordinate import SpectralCoord
```

Comment:
I'm not sure if this is a mis-use of "deprecation warning", but I wonder if we want to add now something like "in the future you will need to use Astropy>=4.1 ? Because at some point we will want to stop maintaining this.

Reply:
Won't we just bump the required version to astropy 4.1 at that point, and that's what users will get when they install specutils?

Reply:
Hmm.  I see your point but I think this is a special case.  That said I don't think we want to block this for that reason, so I guess I'm fine with just merging this without the warning and I can do a separate PR for just this topic.